src/planwise/client/current_project/subs.cljs

```diff
@@ -20,6 +25,42 @@
    (reaction (get-in @db [:current-project :project-data]))))
 
 (register-sub
+ :current-project/read-only?
+ (fn [db [_]]
+   (let [current-data (subscribe [:current-project/current-data])]
+     (reaction (:read-only @current-data)))))
+
+(register-sub
+ :current-project/shares
+ (fn [db [_]]
+   (reaction (get-in @db [:current-project :shares]))))
+
+(register-sub
+ :current-project/shares-search-string
+ (fn [db [_]]
+   (reaction (get-in @db [:current-project :sharing :shares-search-string]))))
+
+(register-sub
+ :current-project/filtered-shares
+ (fn [db [_]]
+   (let [search-string (subscribe [:current-project/shares-search-string])
+         shares (subscribe [:current-project/shares])]
+     (reaction
+       (filterv #(gstring/caseInsensitiveContains (:user-email %) (or @search-string "")) (asdf/value @shares))))))
+
+(register-sub
+ :current-project/share-link
+ (fn [db [_]]
+   (reaction
+     (let [token (get-in @db [:current-project :sharing :token])
+           project-id (get-in @db [:current-project :project-data :id])]
+       (asdf/update token (fn [t]
+                            (str
+                              (.-origin js/document.location)
+                              (routes/project-access {:id project-id
+                                                      :token t}))))))))
```

Comment:
I want this reaction to return an `asdf` with:
- state: same as the token `asdf`
- value: the link created from the token

So the update is used to return a new asdf from the token's one, that preserves the state and modifies the value to return a link. Note that `db` is not modified anywhere, so the reaction could be run any number of times without issues, as the token's asdf is not changed (yay immutability!).